File: VB/CustomParameterEditorAspNetMvcExample/Controllers/HomeController.vb
Imports DevExpress.XtraReports.Native

Public Class HomeController
    Inherits System.Web.Mvc.Controller

    Function Index() As ActionResult
        Return View()
    End Function
    Function Designer() As ActionResult
        Dim report = New DevExpress.XtraReports.UI.XtraReport()
        report.Extensions(SerializationService.Guid) = CustomDataSerializer.Name
        Return View(report)
    End Function
    Function Viewer() As ActionResult
        Return View()
    End Function
End Class

File: CS/CustomParameterEditorAspNetMvcExample/Controllers/HomeController.cs
using DevExpress.XtraReports.Native;
using System.Web.Mvc;

namespace CustomParameterEditorAspNetMvcExample.Controllers
{
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }

        public ActionResult Designer() {
            var report = new DevExpress.XtraReports.UI.XtraReport();
            report.Extensions[SerializationService.Guid] = CustomDataSerializer.Name;
            return View(report);
        }

        public ActionResult Viewer() {
            return View();
        }
    }
}
